### PR TITLE
Remove custom SortedList<T> enumerator

### PR DESCRIPTION
**Pull request by smoogipoo**
The `List<T>` enumerator has additional safety guards preventing against modifying the collection during enumeration. There was no reason for `SortedList<T>` to have a custom enumerator other than to prevent boxing, but `List<T>.Enumerator` is already exposed for that purpose.

### CODE REVIEW

**Comment by bdach** [comment]
Benchmarks, for posterity (`BenchmarkCompositeDrawableAllocations`):

`master`:

|              Method |     Mean |   Error |  StdDev | Gen 0 | Gen 1 | Gen 2 | Allocated |
|-------------------- |---------:|--------:|--------:|------:|------:|------:|----------:|
|   TestEmptyGameLoop | 984.4 us | 2.33 us | 2.18 us |     - |     - |     - |      62 B |
|  TestAllocContainer | 980.1 us | 4.32 us | 4.04 us |     - |     - |     - |    1708 B |
| TestLoadedContainer | 979.8 us | 2.61 us | 2.44 us |     - |     - |     - |    2164 B |
|     TestAllocSprite | 985.1 us | 4.36 us | 4.08 us |     - |     - |     - |    1486 B |
|    TestLoadedSprite | 988.6 us | 1.95 us | 1.82 us |     - |     - |     - |    1749 B |

this PR:

|              Method |     Mean |   Error |  StdDev | Gen 0 | Gen 1 | Gen 2 | Allocated |
|-------------------- |---------:|--------:|--------:|------:|------:|------:|----------:|
|   TestEmptyGameLoop | 980.1 us | 2.51 us | 2.23 us |     - |     - |     - |      43 B |
|  TestAllocContainer | 981.7 us | 6.21 us | 5.81 us |     - |     - |     - |    1726 B |
| TestLoadedContainer | 978.1 us | 2.67 us | 2.50 us |     - |     - |     - |    2165 B |
|     TestAllocSprite | 981.5 us | 5.63 us | 5.27 us |     - |     - |     - |    1467 B |
|    TestLoadedSprite | 980.6 us | 5.44 us | 5.09 us |     - |     - |     - |    1749 B |

Basically no discernible difference.

**Review by bdach — APPROVED** [review]

**Review by peppy — APPROVED** [review]
